x-pack/plugins/data_enhanced/server/search/session/check_running_sessions.test.ts

```diff
@@ -171,6 +182,45 @@ describe('getSearchStatus', () => {
 
       expect(savedObjectsClient.find).toHaveBeenCalledTimes(2);
     });
+
+    test('fetching is abortable', async () => {
+      let i = 0;
+      const abort$ = new Subject();
+      savedObjectsClient.find.mockImplementation(() => {
+        return new Promise((resolve) => {
+          if (++i === 2) {
+            abort$.next();
+          }
+          resolve({
+            saved_objects: i <= 5 ? [emptySO, emptySO, emptySO, emptySO, emptySO] : [],
+            total: 25,
+            page: i,
+          } as any);
+        });
+      });
+
+      await checkRunningSessions$(
+        {
+          savedObjectsClient,
+          client: mockClient,
+          logger: mockLogger,
+        },
+        config
+      )
+        .pipe(takeUntil(abort$))
+        .toPromise();
+
+      // wait a bit before final assertion
+      // to make sure that `find()` isn't called after the test is finished
+      await new Promise((resolve) => {
```

Comment:
It would be great if we could switch to fake timers so we don't actually have to wait 1s here... Is there a reason not to?

Reply:
Improved!